### PR TITLE
add-to-cart test together with necessary page objects for its execution implemented

### PR DESCRIPTION
**Pull request by Ilmaooo**

### CODE REVIEW

File: test/pageobjects/home.page.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        return $('[data-test="product-sort-container"]');
+    }
+
+    // Custom locator function for product name using XPath
+    getProductByName(productName) {
+        return `//*[contains(text(), "${productName}")]`;
+    }
+
+    async clickProductByName(productName) {
+        const productElement = $(this.getProductByName(productName));
```
Comment (Midjoni) [inline test/pageobjects/home.page.js]: You can implement `getProductByName` so it returns an element immediately, so we don't need to wrap it with `$()` every call. Also, `await` is needed.

File: test/pageobjects/product.page.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    //methods
+    async addToCart(){
+        await this.addToCartButton.click();
+    }
+
+    async backToProduct(){
+        await this.backToProductsButton.click();
+    }
+
+    async openCart(){
```
Comment (Midjoni) [inline test/pageobjects/product.page.js]: If this is present in multiple page objects maybe we can think about putting this in a class they extend?

Comment (Ilmaooo) [reply test/pageobjects/product.page.js]: fixed

File: test/pageobjects/home.page.js (hunk 1 above)
Comment (Ilmaooo) [reply test/pageobjects/home.page.js]: fixed i guess

**Review by Midjoni — APPROVED** [review]
LGTM

**Pull request merged**